go/x509/x509.go

```diff
@@ -1123,12 +1228,13 @@ func parseCertificate(in *certificate) (*Certificate, error) {
 					out.IssuingCertificateURL = append(out.IssuingCertificateURL, string(v.Location.Bytes))
 				}
 			}
+		} else {
+			// Unknown extensions are recorded if critical.
+			unhandled = true
 		}
 
-		if e.Critical {
-			// START CT CHANGES
-			nfe.AddError(UnhandledCriticalExtension{e.Id})
-			// END CT CHANGES
+		if e.Critical && unhandled {
```

Comment:
It was intentional; the unhandled critical extensions are now explicitly available in `Certificate.UnhandledCriticalExtensions`, so don't need a (non-fatal) error to mark their presence too.

However, it might be incomplete -- I haven't updated `Verify` to generate a collection of `UnhandledCriticalExtension` errors (nor to use the CT variant of that error that includes the extension OID) -- would that make sense?  (I guess that would need a change to the `Verify` API to also return a non-fatal error list in addition to any successfully generated cert chains).